addons/mail/models/mail_template.py

```diff
@@ -527,10 +528,16 @@ def _generate_template(self, res_ids, render_fields,
 
     @classmethod
     def _parse_partner_to(cls, partner_to):
-        return [
-            int(pid.strip()) for pid in partner_to.split(',')
-            if (pid and pid.strip().isdigit())
-        ]
+        try:
+            partner_to = literal_eval(partner_to or '[]')
+        except (ValueError, SyntaxError):
+            return [
+                int(pid.strip()) for pid in partner_to.split(',')
+                if (pid and pid.strip().isdigit())
+            ]
+        if not isinstance(partner_to, (list, tuple)):
```

Comment:
What about doing both, aka: 1/ make a list (either through literal_eval, either through split) and then 2/ check each item of the list ? that way we combine both worlds, like (untested but to give the idea)
```suggestion
        try:
            # in case of repr or a list / tuple
            partner_to = literal_eval(partner_to or '[]')
        except (ValueError, SyntaxError):
            # comma separated IDs
            partner_to = [partner_to.split(',')]
        if not isinstance(partner_to, (list, tuple)):
            partner_to = [partner_to]
        # check each input is effectively a valid digit for ID
        return [
            int(pid.strip()) for pid in partner_to
            if (pid and pid.strip().isdigit())
        ]
```